packages/trpc/server/routers/viewer/teams/roundRobin/getRoundRobinHostsToReasign.handler.ts

```diff
@@ -35,27 +52,49 @@ export const getRoundRobinHostsToReassign = async ({ ctx, input }: GetRoundRobin
     throw new Error("Booking not found");
   }
 
-  // If input.exclude is "fixedHosts", exclude fixed hosts from the list
-  const excludeFixedHosts = input.exclude === "fixedHosts";
+  // Get event type
+  const eventType = await getEventTypesFromDB(booking.eventType.id);
 
-  const eventTypeHosts = await prisma.host.findMany({
-    where: {
-      eventTypeId: booking.eventType.id,
-      isFixed: excludeFixedHosts ? false : undefined,
-    },
-    select: {
-      user: {
-        select: {
-          id: true,
-          name: true,
-        },
-      },
+  if (!eventType) {
+    throw new Error("Event type not found");
+  }
+
+  const availableEventTypeUsers = eventType.hosts
+    .filter((h) => !h.isFixed && h.user.id !== booking.userId)
+    .map((host) => ({
+      ...host.user,
+      isFixed: host.isFixed,
+      priority: host?.priority ?? 2,
+    }));
+
+  const availableUsers = await ensureAvailableUsers(
+    { ...eventType, users: availableEventTypeUsers as IsFixedAwareUser[] },
+    {
+      dateFrom: dayjs(booking.startTime).format(),
+      dateTo: dayjs(booking.endTime).format(),
+      timeZone: eventType.timeZone || "UTC",
     },
-  });
+    gettingRoundRobinHostsToReassignLogger
+  );
+
+  const availableUserIds = new Set(availableUsers.map((u) => u.id));
+
+  const canViewReassignBusyUsers = isOrgAdmin || (await isTeamAdmin(prisma, ctx.user.id, eventType.teamId));
 
-  const hosts = eventTypeHosts.map((host) => host.user);
+  const roundRobinHostsToReassign = availableEventTypeUsers.reduce((acc, host) => {
+    const status = availableUserIds.has(host.id) ? "available" : "unavailable";
+    if (canViewReassignBusyUsers || status === "available") {
```

Comment:
Easy way to check if the user is an admin else - we only allow the available users into the final list